synthesizer/src/vm/helpers/cost.rs

```diff
@@ -52,10 +52,7 @@ pub fn deployment_cost<N: Network>(deployment: &Deployment<N>) -> Result<(u64, (
 }
 
 /// Returns the *minimum* cost in microcredits to publish the given execution (total cost, (storage cost, namespace cost)).
-pub fn execution_cost<N: Network, C: ConsensusStorage<N>>(
-    vm: &VM<N, C>,
-    execution: &Execution<N>,
-) -> Result<(u64, (u64, u64))> {
+pub fn execution_cost<N: Network, C: ConsensusStorage<N>>(vm: &VM<N, C>, execution: &Execution<N>) -> Result<u64> {
```

Comment:
I actually have to argue against condensing it into one cost. From a usability perspective, people should be able to understand the cost structure & plan their costs. The SDK currently reports all of these costs separately.

I think returning to the original suggestion of having a cost struct is the better idea.

Reply:
Sounds good! 7b3c9bc6f